doc/latex.rst

```diff
@@ -361,35 +328,40 @@ Keys that don't need to be overridden unless in special cases are:
 
    .. versionchanged:: 2.0
       Detection of ``LGR``, ``T2A``, ``X2`` to trigger support of
-      occasional Greek or Cyrillic (``'pdflatex'`` only, as this support
-      is provided natively by ``'platex'`` and only requires suitable
-      font with ``'xelatex'/'lualatex'``).
+      occasional Greek or Cyrillic letters (``'pdflatex'``).
 
    .. versionchanged:: 2.3.0
-      ``'xelatex'`` also executes
+      ``'xelatex'`` executes
       ``\defaultfontfeatures[\rmfamily,\sffamily]{}`` in order to avoid
       contractions of ``--`` into en-dash or transforms of straight quotes
       into curly ones in PDF (in non-literal text paragraphs) despite
       :confval:`smartquotes` being set to ``False``.
 
-``'textgreek'``
-   This is needed for ``pdflatex`` to support Unicode input of Greek
-   letters such as φύσις.  Expert users may want to load the ``textalpha``
-   package with its option ``normalize-symbols``.
+``'fontsubstitution'``
+   Ignored if ``'fontenc'`` was not configured to use ``LGR`` or ``X2`` (or
+   ``T2A``).  In case ``'fontpkg'`` key is configured for usage with some
+   TeX fonts known to be available in the ``LGR`` or ``X2`` encodings, set
+   this one to be the empty string.  Else leave to its default.
 
-   .. hint::
+   Ignored with :confval:`latex_engine` other than ``'pdflatex'``.
```

Comment:
this key contains a dozen lines of LaTeX macros which where previously inserted as part of the default ``'fontpkg'`` key making it difficult for user to customize the key while keeping the feature. Besides this made its way to latex preamble even if not needed, because the macros are conditional. With the refactoring only if user has added LGR of X2 configuration for Greek resp. Cyrillic will these macros end up in latex file.